src/Uno.UI/UI/Xaml/DependencyPropertyDetailsCollection.cs

```diff
@@ -37,184 +50,129 @@ public DependencyPropertyDetailsCollection(Type ownerType, ManagedWeakReference
 
 			var propertiesForType = DependencyProperty.GetPropertiesForType(ownerType);
 
-			var entries = _pool.Rent(propertiesForType.Length);
-
-			for (int i = 0; i < propertiesForType.Length; i++)
+			if (propertiesForType.Length != 0)
 			{
-				ref var entry = ref entries[i];
-				entry.Id = propertiesForType[i].UniqueId;
-				entry.Details = null;
-			}
+				_minId = propertiesForType[0].UniqueId;
+				_maxId = propertiesForType[propertiesForType.Length - 1].UniqueId;
 
-			// Entries are pre-sorted by the DependencyProperty.GetPropertiesForType method
-			AssignEntries(entries, propertiesForType.Length, sort: false);
+				var entriesLength = _maxId - _minId + 1;
+				var entries = _pool.Rent(entriesLength);
```

Comment:
Wondering if it would be interesting to align the `entriesLength` on 16 or 32 to increase the possibility to have a match in the `_pool` and so increase array re-use ?

Reply:
The pool provides arrays based on the closest higher size, and this collection class is already using the larger size if provided :)